### PR TITLE
CloudWatch: Interpolate region in log context query

### LINKED ISSUE #88310: Cloudwatch logs: "show context" doesn't handle region as a variable correctly
### What happened?

If you create a logs dashboard using a template variable for `region`, then click "show context" on a log line, the query that is sent includes the uninterpolated variable instead of the chosen region. No log lines are returned, of course.

### What did you expect to happen?

The region variable should be interpreted in the "show context" query, so that context can be shown.

### Did this work before?

Unknown

### How do we reproduce it?

1. Set up a dashboard with the Logs visualization and Cloudwatch Logs as the data source.
2. Create a dashboard variable for the AWS region
3. Use the dashboard variable for the region setting in the viz query
4. Click on the Show context icon on a line

### Is the bug inside a dashboard panel?

_No response_

### Environment (with versions)?

n/a

### Grafana platform?

I use Grafana Cloud

### Datasource(s)?

cloudwatch

### PR DESCRIPTION
**Pull request by iwysiu**
**What is this feature?**
Interpolates the region in getLogsQueryContext, so that the logs context feature can be used with queries that have a query region

**Which issue(s) does this PR fix?**:

Fixes #88310 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchLogsQueryRunner.ts [hunk 1]
```diff
@@ -180,7 +180,7 @@ export class CloudWatchLogsQueryRunner extends CloudWatchRequest {
       refId: query?.refId || 'A', // dummy
       limit,
       startFromHead: direction !== LogRowContextQueryDirection.Backward,
-      region: query?.region || '',
+      region: this.templateSrv.replace(this.getActualRegion(query?.region)),
```
Comment (nmarrs) [inline public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchLogsQueryRunner.ts:183]: Will this fallback to empty string if it doesn't interpolate correctly / region doesn't exist on the query object?

File: public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchLogsQueryRunner.ts [hunk 2]
```diff
@@ -252,7 +252,7 @@ export class CloudWatchLogsQueryRunner extends CloudWatchRequest {
       this.logQueries[param.refId] = {
         id: param.queryId,
         region: param.region,
-        statsQuery: (param.statsGroups?.length ?? 0) > 0 ?? false,
+        statsQuery: (param.statsGroups?.length ?? 0) > 0,
```
Comment (nmarrs) [inline public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchLogsQueryRunner.ts:255]: Is this related to the bug fix? What is the context for this change?

File: public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchLogsQueryRunner.ts (hunk 1 above)
Comment (iwysiu) [reply public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchLogsQueryRunner.ts:183]: Yup! `getActualRegion` handles the undefined case

File: public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchLogsQueryRunner.ts (hunk 2 above)
Comment (iwysiu) [reply public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchLogsQueryRunner.ts:255]: This was a quick fix for my linter that didn't seem big enough to warrant a separate pr (we never hit the `?? false` portion). I can remove it if we want.

**Review by nmarrs — APPROVED** [review]
